### PR TITLE
Add jQuery removal doc

### PR DESCRIPTION
**Pull request by andysellick**
- provides guidance for removing jQuery from our JavaScript, which is an ongoing task on GOV.UK

Trello card: https://trello.com/c/2T8YA7dv/529-write-down-some-javascript-stuff-make-it-generic

### CODE REVIEW

File: source/manual/how-to-remove-jquery-from-javascript.html.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+```javascript
+var jqExample = $('div')
+jqExample.addClass('test')
+```
+
+Using JS to find an element returns a node, which can have JS functions applied to it e.g.
+
+```javascript
+jsExample = document.querySelector('div')
```
Comment (injms) [inline source/manual/how-to-remove-jquery-from-javascript.html.md:67]:
```suggestion
var jsExample = document.querySelector('div')
```

Comment (andysellick) [reply source/manual/how-to-remove-jquery-from-javascript.html.md]: Done, thanks.

File: source/manual/how-to-remove-jquery-from-javascript.html.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  // do something
+}
+```
+
+### When to use querySelector and querySelectorAll
+
+In jQuery an element can be found on the page using `.find()`. There are two equivalent functions in JS.
+
+- [querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) finds only the first element that matches the given query.
+- [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) finds all elements that match that query, and returns an array of them.
```
Comment (injms) [inline source/manual/how-to-remove-jquery-from-javascript.html.md:114]: I'm going to be picky here - `querySelectorAll()` doesn't return an array but a non-live NodeList. A minor point, as the NodeList can be treated as an array in _most_ instances, but one that I think it worth clarifying as it can trip things up when `forEach()` is used in older browsers.

File: source/manual/how-to-remove-jquery-from-javascript.html.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+Attempting the same thing with JS and `querySelector` will return `null` if no matching element is found, which means any attempt to do anything with that result will cause an error.
+
+This means more checks may have to be added to the code.
+
+```javascript
+var example = document.querySelector('.this')
+
+if (example) {
+  // do something
+}
```
Comment (injms) [inline source/manual/how-to-remove-jquery-from-javascript.html.md:106]: When using `querySelectorAll()` the length of the NodeList needs to be tested rather than the presence as an empty NodeList is still truthy; is it worth mentioning this?

```suggestion
```javascript
var example = document.querySelector('.this')

if (example) {
  // do something
}

var multipleElements = document.querySelectorAll('div')

if (multipleElements.length > 0) {
  // do something
}
```

File: source/manual/how-to-remove-jquery-from-javascript.html.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+jQuery uses the same functions to both get and set attributes on elements.
+
+```javascript
+.attr('class') // replace with getAttribute('class')
+.attr('class', 'value') // replace with setAttribute('class', 'value')
+
+.data('item') // replace with getAttribute('data-item')
+.data('item', 'value') // replace with setAttribute('data-item', 'value')
+```
```
Comment (injms) [inline source/manual/how-to-remove-jquery-from-javascript.html.md:132]: Can the JavaScript equivalents be put in their own code block? This would follow the pattern set in the examples in "Chaining functions" and "jQuery objects are different from JS nodes".

File: source/manual/how-to-remove-jquery-from-javascript.html.md (hunk 2 above)
Comment (andysellick) [reply source/manual/how-to-remove-jquery-from-javascript.html.md]: Updated, thanks.

File: source/manual/how-to-remove-jquery-from-javascript.html.md (hunk 3 above)
Comment (andysellick) [reply source/manual/how-to-remove-jquery-from-javascript.html.md]: Updated, thanks.

File: source/manual/how-to-remove-jquery-from-javascript.html.md (hunk 4 above)
Comment (andysellick) [reply source/manual/how-to-remove-jquery-from-javascript.html.md:136]: As a reference guide I thought it was clearer to keep them together, so readers can easily see the jQuery and JS equivalent. Happy to change to something else though if you can come up with an alternative - I can't think of a better way to do this.

Comment (andysellick) [reply source/manual/how-to-remove-jquery-from-javascript.html.md]: @injms I've had a rethink on this and tried your suggestion - please take a look.

Comment (injms) [reply source/manual/how-to-remove-jquery-from-javascript.html.md]: That's super and exactly what I had in mind - thank you

**Review by injms — APPROVED** [review]
📜 👍
